src/impls.rs

```diff
@@ -5,139 +5,122 @@
+        // a macro to a `struct` generated by a macro
+        $(#[doc = $doc:literal])*
+        $vis:vis struct $name:ident$(<$($gen:ident$(: $bound1:tt $(+ $bound2:tt)*)?),+$(,)?>)? {
+            $(#[$attr1:ident])? $vis1:vis $field1:ident: $type1:ty$(,
+            $(#[$attr2:ident])? $vis2:vis $field2:ident: $type2:ty)*$(,)?
+        }
+    ) => {
+        // build `struct` itself
+        $(#[doc = $doc])*
+        $vis struct $name$(<$($gen$(: $bound1 $(+$bound2)*)?),+>)? {
+            $vis1 $field1: $type1,
+            $($vis2 $field2: $type2),*
+        }
+
+        // implement traits that require specific `where` constraints with the
+        // help of `#[bind]`
+        impl_with_bounds!(
+            $name$(<$($gen$(: $bound1 $(+ $bound2)*)?),+>)?
+            |@$(#$attr1:)? $type1|, $(|@$(#$attr2:)? $type2|,)*
+            core::fmt::Debug => {
+                fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+                    f.debug_struct("$name")
+                    .field("$field1", &self.$field1)
+                    $(.field("$field2", &self.$field2))*
+                    .finish()
+                }
+            },
+            Eq => { },
+            PartialEq => {
+                fn eq(&self, other: &Self) -> bool {
+                    PartialEq::eq(&self.$field1, &other.$field1)
+                    $(&& PartialEq::eq(&self.$field2, &other.$field2))*
+                }
+            },
+            core::hash::Hash => {
+                fn hash<_H: core::hash::Hasher>(&self, state: &mut _H) {
+                    core::hash::Hash::hash(&self.$field1, state);
+                    $(core::hash::Hash::hash(&self.$field2, state);)*
+                }
+            },
+            Clone => {
+                fn clone(&self) -> Self {
+                    Self {
+                        $field1: self.$field1.clone(),
+                        $($field2: self.$field2.clone(),)*
+                    }
+                }
+            },
+        );
+
+        impl$(<$($gen$(: $bound1 $(+ $bound2)*)?),+>)? zeroize::Zeroize for $name$(<$($gen),+>)?
         {
             fn zeroize(&mut self) {
-                impl_zeroize_field_skip_pd!(self, $field1$(, $pd1)?);
-                $(impl_zeroize_field_skip_pd!(self, $field2$(, $pd2)?);)*
+                impl_internal_zeroize!(self, $(#$attr1)? $field1);
+                $(impl_internal_zeroize!(self, $(#$attr2)? $field2);)*
             }
         }
 
-        impl$(<$($gen$(: $bound1$(+ $bound2)*)?),+>)? Drop for $name$(<$($gen),+>)?
+        impl$(<$($gen$(: $bound1 $(+ $bound2)*)?),+>)? Drop for $name$(<$($gen),+>)?
         {
             fn drop(&mut self) {
-                #[allow(unused_imports)]
-                use zeroize::Zeroize;
-                impl_zeroize_field_skip_pd!(self, $field1$(, $pd1)?);
-                $(impl_zeroize_field_skip_pd!(self, $field2$(, $pd2)?);)*
+                zeroize::Zeroize::zeroize(self);
```

Comment:
This should do exactly the same, right?

Reply:
Indeed it does, I confirmed and also pushed a commit on top of this which tests that zeroize is run properly for each of the structs.